--- modules/comment/views_handler_argument_comment_user_uid.inc	2010-12-24 17:48:01.000000000 -0500
+++ modules/comment/views_handler_argument_comment_user_uid.inc	2010-12-24 18:04:51.000000000 -0500
@@ -1,5 +1,5 @@
 <?php
-// $Id: views_handler_argument_comment_user_uid.inc,v 1.2.4.3 2010/11/27 07:51:13 dereine Exp $
+// $Id: views_handler_argument_comment_user_uid.inc,v 1.2.4.4 2010/12/09 23:36:00 dereine Exp $
 
 /**
  * Argument handler to accept a user id to check for nodes that
@@ -36,20 +36,11 @@ class views_handler_argument_comment_use
 
   function query() {
     $this->ensure_my_table();
-
-    $nid_alias = $this->query->add_field('node', 'nid');
-
-    $subselect = db_select('comment', 'c');
-    $subselect->addField('c', 'cid');
-    // @TODO: Figure out why condition/placeholders doesn't work here.
-    $subselect->where("c.uid =  $this->argument");
-    $subselect->where("c.nid = $nid_alias");
-    $subselect = $subselect->countQuery();
-    $subselect->preExecute();
+    $subselect = 'SELECT COUNT(*) FROM comment c WHERE c.uid = '.$this->argument.' AND c.nid = node.nid'; //NOT STANDARD CODE
 
     $condition = db_or()
       ->condition("$this->table_alias.uid", $this->argument, '=')
-      ->where("0 < (" . (string) $subselect . ")");
+      ->where("0 < (" . (string) $subselect . ")"); 
 
     $this->query->add_where(0, $condition);
   }
